backend/contracts/OasisReward.sol

```diff
@@ -1,19 +1,32 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import {ERC721, ERC721Enumerable} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
-import {Base64} from "@openzeppelin/contracts/utils/Base64.sol";
-
+import {
+    ERC721, ERC721Enumerable
+} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import {
+    Base64
+} from "@openzeppelin/contracts/utils/Base64.sol";
+
+/**
+ * @title OasisReward
+ * @notice An ERC721 contract that allows minting of NFT rewards for 
+    Oasis Quizzes.
+ * Contains functions to mint new NFTs, generate SVG images, 
+    and retrieve token data.
+ */
 contract OasisReward is ERC721Enumerable {
-    // Mapping from token ID to IPFS URI
-    mapping(uint256 => string) private _tokenURIs;
+    /// Array of metadata JSON hashes. 
+    /// Used for storage efficient mapping to tokenURI.
```

Comment:
```suggestion
    /// Array of metadata JSON hashes. Used for storage efficient mapping to tokenURI.
```

Reply:
Added Netspec comments. Needs to be 2 lines to avoid comment line limit.